pression/data/Compressor.cpp

```diff
@@ -0,0 +1,107 @@
+
+/* Copyright (c) 2016, Stefan.Eilemann@epfl.ch
+ *
+ * This file is part of Pression <https://github.com/Eyescale/Pression>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#include "Compressor.h"
+
+#include <stdexcept>
+
+namespace pression
+{
+namespace data
+{
+const Compressor::Results& Compressor::compress( const uint8_t* data,
+                                                 size_t size )
+{
+#ifdef PRESSION_USE_OPENMP
+    const size_t chunkSize = getChunkSize();
+    size_t nChunks = size / chunkSize;
+
+    if( nChunks * chunkSize < size )
+        ++nChunks;
```

Comment:
Replace lines 33 to 36 by:

    size_t nChunks = (size + chunkSize - 1) / chunkSize;